tools/probes/probes_main.c

```diff
@@ -199,6 +199,12 @@ void parse_data(char *file_in)
 	}
 
 	packet = malloc(PACKET_MAX_SIZE);
+	if (packet == NULL) {
+		fprintf(stderr, "error: allocation failed, err %d\n",
+			errno);
+		fclose(fd_in);
+		exit(0);
+	}
```

Comment:
You haven't fixed the main issue: 
w_ptr = (uint32_t *)packet;

Reply:
You fixed the issue by removing the packed attribute, that's really odd...

Reply:
There is nothing wrong with assigning a pointer but it can if there is alignment inconsistency and the error was clear:
error: converting a **packed** ‘struct probe_data_packet’ pointer (**alignment 1**) to a ‘uint32_t’ {aka ‘unsigned int’} pointer (**alignment 4**) may result in an unaligned pointer value

packed attribute defines struct members aligment and this is the true reason for alignment inconsistency

Reply:
@abonislawski please use ``` __builtin_assume_aligned(ptr, alignment)```

Reply:
@lgirdwood can you explain why? Without "packed" we have both alignments equal, no 
assumptions = no errors

Reply:
Data must be packed between FW, kernel and userspace since they may all use different compilers that will optimise and can pack differently.

Reply:
How about `__attribute__((packed, aligned(4)))`? Does that avoid the warning?

Both `aligned(4)` _and_ `packed` because in the surprising case of a compiler aligning uint32_t on 64 bits, I suspect gcc's `aligned(4)` alone behaves like the newer C11 `_Alignas(4)`  which refuses to  "weaken the alignment the type would have had naturally" https://en.cppreference.com/w/c/language/_Alignas

This page claims that `packed` (conceptually ~= `force_aligned(1)` ) can cause slower code http://digitalvampire.org/blog/index.php/2006/07/31/why-you-shouldnt-use-__attribute__packed/
It would be interesting to disassemble and compare.

https://groups.google.com/forum/#!msg/x86-64-abi/IT0wgB6Fd5g/awOmKV7UBgAJ

Reply:
@lgirdwood ok seems reasonable but is there a real case with the struct like that? Cant imagine other than future 64bits, I will update with alignment assume

@marc-hb yeap, packed and aligned 4 looks the best, great for now (because actually doing nothing) and ready for 64b.
As you said alignment 1 can be slower, not sure if even here when forced 1 but in real its still 4, probably still slower because compiler cannot assume align4